libraries/botframework-schema/src/ActivityEx.ts

```diff
@@ -0,0 +1,342 @@
+
+    export function AsTypingActivity(source: Partial<Activity>): ITypingActivity {
+        return isActivity(source, ActivityTypes.Typing) ? this : null;
+    }
+
+    export function AsEndOfConversationActivity(source: Partial<Activity>): IEndOfConversationActivity {
+        return isActivity(source, ActivityTypes.EndOfConversation) ? this : null;
+    }
+
+    export function AsEventActivity(source: Partial<Activity>): IEventActivity {
+        return isActivity(source, ActivityTypes.Event) ? this : null;
+    }
+
+    export function AsInvokeActivity(source: Partial<Activity>): IInvokeActivity {
+        return isActivity(source, ActivityTypes.Invoke) ? this : null;
+    }
+
+    export function AsMessageUpdateActivity(source: Partial<Activity>): IMessageUpdateActivity {
+        return isActivity(source, ActivityTypes.MessageUpdate) ? this : null;
+    }
+
+    export function AsMessageDeleteActivity(source: Partial<Activity>): IMessageDeleteActivity {
+        return isActivity(source, ActivityTypes.MessageDelete) ? this : null;
+    }
+
+    export function AsMessageReactionActivity(source: Partial<Activity>): IMessageReactionActivity {
+        return isActivity(source, ActivityTypes.MessageReaction) ? this : null;
+    }
+
+    export function AsSuggestionActivity(source: Partial<Activity>): ISuggestionActivity {
+        return isActivity(source, ActivityTypes.Suggestion) ? this : null;
+    }
+
+    export function AsTraceActivity(source: Partial<Activity>): ITraceActivity {
+        return isActivity(source, ActivityTypes.Trace) ? this : null;
+    }
+
+    export function AsHandoffActivity(source: Partial<Activity>): IHandoffActivity {
+        return isActivity(source, ActivityTypes.Handoff) ? this : null;
+    }
+
+    export function HasContent(source: Partial<Activity>): boolean  {
+        if(source.text) {
+            return true;
+        }
+
+        if(source.summary) {
+            return true;
+        }
+
+        if(source.attachments) {
+            return true;
+        }
+
+        if(source.channelData) {
+            return true;
+        }
+
+        return false;
+    }
+
+    export function GetMentions(source: Partial<Activity>) : Mention[] {
+        return source.entities.filter(x => x.type.toLowerCase() === 'mention')
+        .map(e => e as Mention );
+    }
+
+    export function GetConversationReference(source: Partial<Activity>) : Partial<ConversationReference> {
+        return {
+            activityId: source.id,
+            user: source.from,
+            bot: source.recipient,
+            conversation: source.conversation,
+            channelId: source.channelId,
+            locale: source.locale,
+            serviceUrl: source.serviceUrl
+        } as ConversationReference;
+    }
+
+    export function GetReplyConversationReference(source: Partial<Activity>, reply: Partial<ResourceResponse>) : Partial<ConversationReference> {
+        let reference = GetConversationReference(source);

```

Comment:
The variable could be a `const`, as its entire value doesn't change.

Reply:
Feedback applied. Thanks!